libraries/AP_HAL_ChibiOS/UARTDriver.h

```diff
@@ -144,6 +144,9 @@ class ChibiOS::UARTDriver : public AP_HAL::UARTDriver {
      */
     bool is_dma_enabled() const override { return rx_dma_enabled && tx_dma_enabled; }
 
+    // get the last time when the tx write buffer got emptied
+    uint32_t get_last_tx_empty_us() const override { return (tx_dma_enabled && _writebuf.is_empty()) ? _last_write_completed_us : 0; }
```

Comment:
why this depends on DMA config ?

Reply:
Because the _last_write_completed_us only gets written to if tx_dma_enabled == true